backend/diagnostics.go

```diff
@@ -83,3 +90,96 @@ func (fn CollectMetricsHandlerFunc) CollectMetrics(ctx context.Context) (*Collec
 type CollectMetricsResult struct {
 	PrometheusMetrics []byte
 }
+
+// ToUsageStats filters a CollectMetricsResult and returns usage stats
+func (res *CollectMetricsResult) ToUsageStats(pluginID string) map[string]interface{} {
+	var (
+		parser expfmt.TextParser
+		r      = regexp.MustCompile(`usage_stats[_]?`)
+		stats  = map[string]interface{}{}
+		reader = bytes.NewReader(res.PrometheusMetrics)
+	)
+
+	mfs, err := parser.TextToMetricFamilies(reader)
+	if err != nil {
+		Logger.Error("unable to parse usage stats from metric", "plugin", pluginID, "err", err)
+	}
+	for _, mf := range mfs {
+		if !strings.HasPrefix(mf.GetName(), "usage_stats") {
+			continue
+		}
+		name := r.ReplaceAllString(mf.GetName(), "")
+		for _, m := range mf.Metric {
+			labels := []string{pluginID, name}
+			for _, l := range m.Label {
+				labels = append(labels, l.GetName(), l.GetValue())
+			}
+			converted := map[string]interface{}{}
+			switch mf.GetType() {
+			case prom.MetricType_COUNTER:
+				converted = convertCounter(labels, m)
+			case prom.MetricType_GAUGE:
+				converted = convertGauge(labels, m)
+			case prom.MetricType_SUMMARY:
+				converted = convertSummary(labels, m)
+			case prom.MetricType_HISTOGRAM:
+				converted = convertHistogram(labels, m)
+			}
+			for k, v := range converted {
+				stats[k] = v
+			}
+		}
+	}
+
+	return stats
+}
+
+func convertCounter(labels []string, m *prom.Metric) map[string]interface{} {
+	stats := map[string]interface{}{}
+	key := toUsageStatKey(labels...)
+	stats[key] = m.GetCounter().GetValue()
+	return stats
+}
+
+func convertGauge(labels []string, m *prom.Metric) map[string]interface{} {
+	stats := map[string]interface{}{}
+	key := toUsageStatKey(labels...)
+	stats[key] = m.GetGauge().GetValue()
+	return stats
+}
+
+func convertSummary(labels []string, m *prom.Metric) map[string]interface{} {
+	stats := map[string]interface{}{}
+	summary := m.GetSummary()
+	key := toUsageStatKey(append(labels, "sum")...)
+	stats[key] = summary.GetSampleSum()
+	key = toUsageStatKey(append(labels, "count")...)
+	stats[key] = float64(summary.GetSampleCount())
+	for _, q := range summary.Quantile {
+		key = toUsageStatKey(append(labels, "quantile", fmt.Sprintf("%.2f", q.GetQuantile()))...)
+		stats[key] = q.GetValue()
+	}
+	return stats
+}
+
+func convertHistogram(labels []string, m *prom.Metric) map[string]interface{} {
+	h := m.GetHistogram()
+	stats := map[string]interface{}{}
+	key := toUsageStatKey(append(labels, "sum")...)
+	stats[key] = h.GetSampleSum()
+	key = toUsageStatKey(append(labels, "count")...)
+	stats[key] = float64(h.GetSampleCount())
+	for _, b := range h.Bucket {
+		key = toUsageStatKey(append(labels, "bucket", fmt.Sprintf("%.0f", b.GetUpperBound()))...)
+		stats[key] = float64(b.GetCumulativeCount())
+	}
+	return stats
+}
```

Comment:
Summary and histogram have a related problem in that they're going to just be snapshots of the most recent scrape interval at the moment the report is sent, which is going to give us inaccurate data if the behavior being measured varies throughout the day.